stistools/poisson_err.py

```
@@ -0,0 +1,146 @@
+Gaussian appoximation. This re-calculation can be important for datasets that have low detector counts
+(e.g., FUV continuum).
+
+The input file for :func:`poisson_err` is a STIS x1d file. :func:`poisson_err` reads in the x1d file,
+calculates the correct 1-sigma upper and lower Poisson confidence interval limits,
+and creates new columns in the fits file.
+
+Because we use NET_ERROR to back out the pre-dark-subtracted counts, we are assuming that err=sqrt(I), which
+is valid only for MAMA observations. CCD observations will also have gain, bias, and readnoise corrections.
+
+File will be written to the filename provided by "output", creating a new file or overwriting an existing one.
+
+Examples
+--------
+
+:func:`poisson_err` with default values:
+
+>>> import stistools
+>>> stistools.poisson_err.poisson_err("ld9m12erq_x1d.fits","ld9m12erq_poisson_x1d.fits")
+
+"""
+
+__taskname__ = "poisson_err"
+__version__ = "1.0"
+__vdate__ = "23-August-2024"
+__author__ = "J. Lothringer"
+
+
+def poisson_err(x1dfile, output,  verbose=True):
+    """Adds Poisson Confidence Interval columns to extracted spectra.
+       For use with MAMA data only. Works with echelle or first-order
+       spectra.
+
+        Parameters
+        ----------
+        x1dfile: str
+            Input MAMA x1d file from which we will calculate the poisson
+            confidence interval.
+
+        output: str
+            Name of the output FITS file. Will overwrite existing file.
+    
+        verbose: bool
+            Prints a completion message before function returns.
+
+        Returns
+        -------
+        Nothing is returned directly, but the file is written to output.
+    """
+    #Open the file
+    with fits.open(x1dfile) as hdu:
+        #Read in primary header
+        header = hdu[0].header
+
+        #Spit out warning if using CCD data
+        if 'CCD' in header['DETECTOR']:
```

We should raise an exception instead of printing a warning and continuing.

Here's a way to do it that is robust to the absence of header keywords:
```python
    try:
        instrument = hdu[0].header['INSTRUME'].strip()
        detector = hdu[0].header['DETECTOR'].strip()
        if (instrument != 'STIS') and (detector not in {'NUV-MAMA', 'FUV-MAMA'}:
            raise ValueError
    except (ValueError, IndexError,):
        raise ValueError(f"{x1dfile}: Not a STIS/MAMA observation.")
```
(You also won't need the `header = hdu[0].header` line above if accessed this way.)

---

Replaced printed warning with a raised exception.